plugin/src/main/java/org/opensearch/ml/task/MLTaskManager.java

```diff
@@ -37,6 +38,7 @@
  * MLTaskManager is responsible for managing MLTask.
  */
 @Log4j2
+@Getter
```

Comment:
This looks risky to expose `taskCaches` field. As you can see we have several `synchronized` methods. By exposing `taskCaches`, anyone can add/remove tasks from it.

Reply:
This is a good point. Will remove @Getter add a public method in the MLTaskManager to return cached tasks in arrays.